src/main/java/com/denizenscript/depenizen/bukkit/commands/quests/QuestsCommand.java

```diff
@@ -0,0 +1,221 @@
+
+        // Iterate through arguments
+        for (Argument arg : scriptEntry.getProcessedArgs()) {
+
+            if (!scriptEntry.hasObject("action")
+                    && arg.matchesEnum(Action.values())) {
+                scriptEntry.addObject("action", arg.asElement());
+            }
+
+            else if (!scriptEntry.hasObject("quest-id")
+                    && arg.matchesPrefix("quest", "quest-id")) {
+                scriptEntry.addObject("quest-id", arg.asElement());
+            }
+            
+            else if (!scriptEntry.hasObject("stage-no")
+                    && arg.matchesPrefix("stage", "stage-no")
+                    && arg.matchesPrimitive(ArgumentHelper.PrimitiveType.Integer)) {
+                scriptEntry.addObject("stage-no", arg.asElement());
+            }
+            
+            else if (!scriptEntry.hasObject("points")
+                    && arg.matchesPrefix("points")
+                    && arg.matchesPrimitive(ArgumentHelper.PrimitiveType.Integer)) {
+                scriptEntry.addObject("points", arg.asElement());
+            }
+            
+            else if (!scriptEntry.hasObject("state")
+                    && arg.matchesPrefix("state")
+                    && arg.matchesEnum(State.values())) {
+                scriptEntry.addObject("state", arg.asElement());
+            }
+            else {
+                arg.reportUnhandled();
+            }
+
+        }
+
+        if (!scriptEntry.hasObject("action")) {
+            throw new InvalidArgumentsException("Must specify a valid action!");
+        }
+        
+        scriptEntry.defaultObject("state", new ElementTag("TRUE"));
+    }
+
+    @Override
+    public void execute(ScriptEntry scriptEntry) {
+        Quests quests = (Quests) QuestsBridge.instance.plugin;
+        
+        BukkitScriptEntryData scriptEntryData = (BukkitScriptEntryData) scriptEntry.entryData;
+
+        // Get objects
+        ElementTag action = scriptEntry.getElement("action");
+        ElementTag questId = scriptEntry.getElement("quest-id");
+        ElementTag stageNum = scriptEntry.getElement("stage-no");
+        ElementTag points = scriptEntry.getElement("points");
+        ElementTag state = scriptEntry.getElement("state");
+
+        PlayerTag player = scriptEntryData.getPlayer();
+
+        // Report to dB
+        Debug.report(scriptEntry, getName(), action.debug() 
+                + (questId != null ? questId.debug() : "") 
+                + (stageNum != null ? stageNum.debug() : "") 
+                + (points != null ? points.debug() : "") 
+                + (state != null ? state.debug() : ""));
+
+        switch (Action.valueOf(action.asString().toUpperCase())) {
+
+            case ADD: {
+
+                if (questId != null && state != null) {
+                    for (Quest quest : quests.getQuests()) {
+                        if (quest.getId().equals(questId.asString()) && player != null) {

```

Comment:
Should a null player _ever_ be valid for this command? This might be better handled as a single check of `hasPlayer` on the entry data in `parseArgs`, rather than null checks throughout.

Reply:
I wasn't sure, but to be honest I don't feel comfortable making that change without taking a look at your utility class, which I don't currently have time to do. Feel free to make the necessary adjustments :)

Reply:
```
if (!Utilities.entryHasPlayer(scriptEntry)) {
    throw new InvalidArgumentsException("Must have a linked player!");
}
```

Reply:
That throw would need to be surrounded by a try/catch.

Reply:
It very specifically would not be if included in the parseArgs as I said to do, as you can see it specifically declares it: `public void parseArgs(ScriptEntry scriptEntry) throws InvalidArgumentsException {`

Reply:
Indeed, my mistake.